### PR TITLE
custom faucet hosts and various typing fixes for NFT support that were originally missed

### PR DESCRIPTION
**Pull request by ledhed2222**
## High Level Overview of Change

* Various fixes to NFT implementation
* Allow users to specify custom faucet hosts
* Run docgen and commit changes

### Context of Change

See linked issue for rationale on specifying custom host.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates

### CODE REVIEW

File: packages/xrpl/src/Wallet/fundWallet.ts [hunk 1]
```diff
@@ -39,14 +39,23 @@ const MAX_ATTEMPTS = 20
...
  *
  * @param this - Client.
- * @param wallet - An existing XRPL Wallet to fund, if undefined, a new Wallet will be created.
+ * @param wallet - An existing XRPL Wallet to fund. If undefined or null, a new Wallet will be created.
+ * @param options - See below.
+ * @param options.faucetHost - A custom host for a faucet server. On devnet and
+ * testnet, `fundWallet` will attempt to determine the correct server
+ * automatically. In other environments, or if you would like to customize the
+ * faucet host in devnet or testnet, you should provide the host using this
+ * option.
```
Comment (JST5000) [inline packages/xrpl/src/Wallet/fundWallet.ts]: It might be nice to include an example to clarify if the port is needed, if https://... is needed, etc.

**Review by JST5000 — APPROVED** [review]
LGTM!

**Comment by ledhed2222** [comment]
> Can the docgen changes be in a separate PR, to make this neater/easier to review?

Those changes were made here so that I could validate the correctness of the docs changes made for faucet generation, so not entirely unrelated. At this point I'd rather not separate.

**Review by khancode — APPROVED** [review]
LGTM

**Pull request merged**